app/assets/javascripts/discourse/app/lib/transform-post.js

```diff
@@ -37,6 +38,9 @@ export function transformBasicPost(post) {
     user_id: post.user_id,
     usernameUrl: userPath(post.username),
     username: post.username,
+    badgesGranted:
+      post.badges_granted &&
+      post.badges_granted.map((badge) => Badge.createFromJson(badge)[0]),
```

Comment:
null operator for a bit of code golf

```suggestion
      post.badges_granted?.map((badge) => Badge.createFromJson(badge)[0]),
```